parlai/crowdsourcing/tasks/turn_annotations_static/turn_annotations_blueprint.py

```diff
@@ -0,0 +1,190 @@
+        """
+        super().add_args_to_group(group)
+        group.add_argument(
+            "--subtasks-per-unit",
+            dest="subtasks_per_unit",
+            type=int,
+            default=-1,
+            help="number of subtasks/comparisons to do per unit",
+        )
+        group.add_argument(
+            "--annotate-last-utterance-only",
+            dest="annotate_last_utterance_only",
+            type=bool,
+            default=False,
+            help="If we only want the crowdworker to annotate the last utterance in the conversation",
+        )
+        group.add_argument(
+            "--onboarding-data",
+            dest="onboarding_data",
+            type=str,
+            default='./json/onboarding.json',
+            help="Path to data and answers for onboarding task in JSON format",
+        )
+        group.add_argument(
+            "--annotation-buckets",
+            dest="annotation_buckets",
+            type=str,
+            default='./json/annotation_buckets.json',
+            help="As per Turn Annotations task, path to annotation buckets which will be checkboxes for worker to annotate an utterance.",
+        )
+
+    def get_frontend_args(self) -> Dict[str, Any]:
+        """
+        Specifies what options within a task_config should be forwarded to the client
+        for use by the task's frontend.
+        """
+        with open(self.opts['onboarding_data'], "r", encoding="utf-8-sig") as f:
+            onboarding_data = json.loads(f.read())
+
+        with open(self.opts['annotation_buckets'], "r", encoding="utf-8-sig") as f:
+            annotation_buckets = json.loads(f.read())
+
+        return {
+            "task_description": self.opts['task_description'],
+            "task_title": self.opts['task_title'],
+            "onboarding_data": onboarding_data,
+            "annotation_buckets": annotation_buckets,
+            "annotate_last_utterance_only": self.opts['annotate_last_utterance_only'],
+            "frame_height": '100%',
+            "num_subtasks": self.opts["subtasks_per_unit"],
+            "block_mobile": True,
+        }
+
+    def process_data(self, data_dicts):
+        """
+        Override this in a subclass if you want to change how data is processed from
+        input file before being sent to the frontend.
+        """
+        output = []
+        for d in data_dicts:
+            new_dialogue = []
+            for utt in d['dialog']:
+                # If there is a persona, which is context as in the ConvAI2
+                # task, we don't want to display the persona utterances
+                if 'persona' not in utt[0]['text']:
+                    new_dialogue.append({'text': utt[0]['text'], 'agent_idx': 0})
+                if 'persona' not in utt[1]['text']:
+                    new_dialogue.append({'text': utt[1]['text'], 'agent_idx': 1})
+            output.append(new_dialogue)
+        return output
```

Comment:
it would be nice if we can shuffle the data before return it.

Reply:
I actually didn't shuffle it for transparency's sake. Maybe somebody can add that in the future.